alws/crud/build_node.py

```diff
@@ -93,6 +93,24 @@ def generate_restart_tasks_query(build_id: int, parallel: bool = False):
     return query
 
 
+async def __drop_srpm_if_all_arches_failed(db: Session, task: models.BuildTask):
+    tasks_with_same_ref = (await db.execute(select(
+        models.BuildTask).where(
+            models.BuildTask.build_id == task.build_id,
+            models.BuildTask.ref_id == task.ref_id
+        )
+    )).scalars().all()
```

Comment:
You're right, and I first implemented it that way, but looked a bit weird and I thought it could be a good idea to split it into a smaller function, even if we're performing an additional query per project. I'll try differently and see if I get to something decent.

Reply:
Fixed this too, let me know if looks better to you